--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -397,7 +397,7 @@
     PREFIX libuv
     GIT_REPOSITORY https://github.com/libuv/libuv
     # Sync version with flake.nix
-    GIT_TAG v1.48.0
+    GIT_TAG e9f29cb984231524e3931aa0ae2c5dae1a32884e
     CMAKE_ARGS
       -DCMAKE_BUILD_TYPE=Release -DLIBUV_BUILD_TESTS=OFF -DLIBUV_BUILD_SHARED=OFF -DCMAKE_AR=${CMAKE_AR}
       -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE} -DCMAKE_POSITION_INDEPENDENT_CODE=ON
